### ISSUE TITLE
Partially expand issue numbers to issue titles

### ISSUE THREAD

**Issue by thomaseizinger** [waffleio/waffle.io#3826]
Currently, when an issue is blocked or blocks another one, waffle only displays the issue number on the board:

![image](https://user-images.githubusercontent.com/5486389/52096334-a0c85700-261a-11e9-803a-08ab9c38616d.png)

We constantly run into the problem that we are visually searching the board to see, where this item is, just to know which one it is.

There should be a way to allow waffle to expand this issue number to parts of the title. Depending on the board, this might look very ugly though. For example, waffle board of waffle.io has many columns, so there is actually no space to this display this. However, this is not the case for all boards.

In our grooming sessions, our board usually looks like this:

![image](https://user-images.githubusercontent.com/5486389/52096501-22b88000-261b-11e9-95b9-ed592c4d925f.png)

As you can see, there would be plenty of space to at least partially display the issue title.

I'd think this would a great opportunity for media-queries. Based on the actual width of the card, the issue title could be shown next the number.

**Comment by marydavis** [comment]
Hey @thomaseizinger what about a tooltip that shows the entire title on hover? Would that work for you?

**Comment by thomaseizinger** [comment]
Hey @marydavis,

thanks for picking this up!

I think a tooltip would work, as long as it loads instantly :)
Our aim usually is to locate the other issue on the board: something we currently achieve by doing CTRL+F for the issue number.
In addition to the tooltip, could you make the other card flash or highlight on hover?
I think that would be the killer feature to locate related cards :)

**Comment by marydavis** [comment]
Thanks @thomaseizinger for the feedback :) Awesome, we will probably start with the tooltip and see how that goes. In regards to highlighting the card, if you click on 'blocked by #xyz' the board will filter by the blocking card - does that meet your needs of finding it?

**Comment by thomaseizinger** [comment]
> In regards to highlighting the card, if you click on 'blocked by #xyz' the board will filter by the blocking card - does that meet your needs of finding it?

I didn't know this!
Just tried but the result was a bit disappoint to be honest :(

1. It shows all issues that are somehow *related* to the one I clicked one
2. Pressing the back button of my mouse goes back to the previous URL but doesn't reset the filter on the board.

Overall, the tooltip and maybe flashing would be great because that could happen instantly.
Our board has > 10 repositories connected and everytime we need to *click* on something to get more information, it loads for at least a full second or more.

This is especially painful because the dialog that comes up doesn't give you more information about related issues. You have to leave waffle and go to GitHub to see which issues are actually referenced.

**Comment by marydavis** [comment]
Hey @thomaseizinger

We just shipped the tooltip addition, so when you hover over an issue number in the dependency list on a card you will now see the entire card title in the tooltip. Hope this is at least one step in the right direction :)

In regards to the filter when clicking on the dependency, there are two easy ways to remove that filter without having to refresh the board. First, you can click the dependency again which will remove the filter or pressing `shift + f` is a keyboard shortcut to remove all filters.

Boards have a lot of data, so full refreshes can be taxing on performance unfortunately. Although ideally with realtime, you shouldn't have to do a full refresh often. You mentioned that when you click on something to get more information, it loads for at least a full second or more - are you referring to the issue modal taking a full second or more to show or something else?

**Comment by thomaseizinger** [comment]
> We just shipped the tooltip addition, so when you hover over an issue number in the dependency list on a card you will now see the entire card title in the tooltip. Hope this is at least one step in the right direction :)

That was quick! Thanks a lot :)

> You mentioned that when you click on something to get more information, it loads for at least a full second or more - are you referring to the issue modal taking a full second or more to show or something else?

Yes, I am referring to the time the modal takes to open. For our concrete issue (finding dependent cards), opening the model doesn't help much though because:

1. It doesn't display reverse dependencies (another issue *requiring* this one)
2. It also just shows the text like "Required by #XYZ." (without the title) and clicking on that takes you to GitHub anyway.

We often find ourselves doing the following:

- Identify a card that has high priority but having a dependency that says: `Requires #XYZ.`
- We ask ourselves: "What is XYZ?" because we have to do that one first apparently.
- We click on the modal so that we can click on the issue link (#XYZ). This takes us GitHub where we can finally see what the blocking issue is.

There are several reasons for this:
1. Looking at the card itself does not reveal what the issue is: the tooltip helps greatly with that!)
2. Just from the card, you cannot directly navigate to the issue on GitHub: not a big problem IMO since we would like to minimize going to GitHub anyway (additional loading time, loosing focus of the board, etc)
3. Filtering the board by clicking on the issue is sub-optimal aswell:
    1. While the filtering is instant, going back is not. There is no loading indicator what so ever and it takes a second or two to clear the filter.
    2. Filtering the board only shows all related issues. Could be 2, could also be 8. Within the filtered board, you have to search again manually with your eyes.
    3. Filtering the board makes you loose complete focus because all the other issues are gone.

I really appreciate the tooltip!
Maybe there are even better ways on how this usecase can be solved :)

> Boards have a lot of data, so full refreshes can be taxing on performance unfortunately. Although ideally with realtime, you shouldn't have to do a full refresh often.

We are trying to avoid these anyway, that is why realtime features like the tooltip help greatly!
